subsys/shell/shell_uart.c

```diff
@@ -155,18 +155,14 @@ static int uninit(const struct shell_transport *transport)
 	return 0;
 }
 
-static int enable(const struct shell_transport *transport, bool blocking)
+static int enable(const struct shell_transport *transport, bool blocking_tx)
 {
 	const struct shell_uart *sh_uart = (struct shell_uart *)transport->ctx;
 
-	sh_uart->ctrl_blk->blocking = blocking;
+	sh_uart->ctrl_blk->blocking_tx = blocking_tx;
 
-	if (blocking) {
-		if (!IS_ENABLED(CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN)) {
-			k_timer_stop(sh_uart->timer);
-		}
+	if (blocking_tx) {
 #ifdef CONFIG_SHELL_BACKEND_SERIAL_INTERRUPT_DRIVEN
-		uart_irq_rx_disable(sh_uart->ctrl_blk->dev);
```

Comment:
Why this line is removed?

Reply:
because this mode is now blocking TX. we only care about TX actually.

Reply:
I've added comment to shell.h